diff --git a/vtysh/utils/passwd_srv_utils.c b/vtysh/utils/passwd_srv_utils.c
index ae830e5..896747e 100644
--- a/vtysh/utils/passwd_srv_utils.c
+++ b/vtysh/utils/passwd_srv_utils.c
@@ -63,7 +63,10 @@ int passwd_srv_path_manager_init(void)
     lt_dlerror();
 
     /* open shared object to be used */
-    so_handle = lt_dlopen(PASSWD_SRV_SO_LIB);
+    char lib_path[PASSWD_SRV_MAX_STR_SIZE];
+    snprintf(lib_path, sizeof(lib_path), "%s" PASSWD_SRV_SO_LIB,
+             (getenv("OPENSWITCH_INSTALL_PATH") ? getenv("OPENSWITCH_INSTALL_PATH") : ""));
+    so_handle = lt_dlopen(lib_path);
 
     if (lt_dlerror())
     {
diff --git a/vtysh/vtysh.c b/vtysh/vtysh.c
index 6f113be..b75f5f1 100755
--- a/vtysh/vtysh.c
+++ b/vtysh/vtysh.c
@@ -155,6 +155,8 @@ bool rbac_check_user_permission(char * username, char * permission)
     lt_dlhandle dhhandle = 0;
     bool retval = 0;
     bool (*fun_ptr)(const char *, const char *) = NULL;
+    char libPath[VTYSH_MAX_PATHNAME_LEN+1];
+
     if ((username == NULL) || (permission == NULL))
     {
        return false;
@@ -162,7 +164,10 @@ bool rbac_check_user_permission(char * username, char * permission)
     lt_dlinit();
     lt_dlerror();
 
-    dhhandle = lt_dlopen ("/usr/lib/librbac.so.0.1.0");
+    snprintf(libPath, sizeof(libPath), "%s/usr/lib/librbac.so.0.1.0",
+             (getenv("OPENSWITCH_INSTALL_PATH") ? getenv("OPENSWITCH_INSTALL_PATH") : ""));
+
+    dhhandle = lt_dlopen (libPath);
 
     if (lt_dlerror())
     {
@@ -2328,27 +2333,52 @@ execute_command (const char *command, int argc, const char *arg[])
    return ret;
 }
 
+static char *temporary_startup_db = NULL;
+static char *temporary_startup_db_lock = NULL;
+static char *temporary_process_pid = NULL;
+static char *ovsdb_path = NULL;
+static char *temporary_server_pidfile = NULL;
+static char *temporary_server_socket = NULL;
+void
+init_temp_db(void)
+{
+    char *dataroot = getenv("OPENSWITCH_DATA_PATH");
+    if (!dataroot)
+        dataroot = "";
+    if (!temporary_startup_db)
+        temporary_startup_db = xasprintf("%s" TEMPORARY_STARTUP_DB, dataroot);
+    if (!temporary_startup_db_lock)
+        temporary_startup_db_lock = xasprintf("%s" TEMPORARY_STARTUP_DB_LOCK, dataroot);
+    if (!temporary_process_pid)
+        temporary_process_pid = xasprintf("%s" TEMPORARY_PROCESS_PID, dataroot);
+    if (!ovsdb_path)
+       ovsdb_path = xasprintf("%s" OVSDB_PATH, dataroot);
+    if (!temporary_server_pidfile)
+       temporary_server_pidfile = xasprintf("--pidfile=%s", temporary_process_pid);
+    if (!temporary_server_socket)
+       temporary_server_socket = xasprintf("punix:%s/var/run/openvswitch/" TEMPORARY_STARTUP_SOCKET, dataroot);
+}
+
 int
 remove_temp_db(int initialize)
 {
-    char file_name[]= TEMPORARY_PROCESS_PID;
-    char *remove_db[] = {TEMPORARY_STARTUP_DB};
+    char *remove_db[] = {temporary_startup_db};
     FILE * fp;
     char *line = NULL;
     size_t len = 0;
     int file_not_present = 0;
 
-    fp = fopen(file_name,"r");
+    fp = fopen(temporary_process_pid,"r");
     if (fp == NULL)
     {
         if (initialize == 0)
         {
-            VLOG_ERR("Error while opening %s file\n", file_name);
+            VLOG_ERR("Error while opening %s file\n", temporary_process_pid);
             return -1;
         }
         else
         {
-            VLOG_INFO("No %s file present\n",file_name);
+            VLOG_INFO("No %s file present\n",temporary_process_pid);
             file_not_present = 1;
         }
     }
@@ -2366,9 +2396,9 @@ remove_temp_db(int initialize)
         fclose(fp);
     }
 
-    if (access(TEMPORARY_STARTUP_DB, F_OK) == -1)
+    if (access(temporary_startup_db, F_OK) == -1)
     {
-        VLOG_INFO("No %s file present\n", TEMPORARY_STARTUP_DB);
+        VLOG_INFO("No %s file present\n", temporary_startup_db);
     }
     else
     {
@@ -2389,11 +2419,19 @@ DEFUN (show_startup_config,
 {
   char *arguments[] = {"show", "startup-config", "cli"};
   char *temp_args[] = {"-D", TEMPORARY_STARTUP_SOCKET, "-c", "show running-config "};
-  char *copy_db[] = {OVSDB_PATH, TEMPORARY_STARTUP_DB};
-  char *run_server[] = {"--pidfile=/var/run/openvswitch/temp_startup.pid", "--detach", "--remote", "punix:/var/run/openvswitch/temp_startup.sock", TEMPORARY_STARTUP_DB};
-  char *remove_tempstartup_db[] = {"-f", TEMPORARY_STARTUP_DB_LOCK};
+  char *copy_db[] = {NULL, NULL};
+  char *run_server[] = {NULL, "--detach", "--remote", NULL, NULL};
+  char *remove_tempstartup_db[] = {"-f", NULL};
   int ret = 0;
 
+  init_temp_db();
+  copy_db[0] = ovsdb_path;
+  copy_db[1] = temporary_startup_db;
+  run_server[0] = temporary_server_pidfile;
+  run_server[3] = temporary_server_socket;
+  run_server[4] = temporary_startup_db;
+  remove_tempstartup_db[1] = temporary_startup_db_lock;
+
   // Check if temporary DB exists and OVSDB server running. If yes, remove it.
   remove_temp_db(1);
 
diff --git a/vtysh/vtysh.h b/vtysh/vtysh.h
index 498b37b..95f51fc 100644
--- a/vtysh/vtysh.h
+++ b/vtysh/vtysh.h
@@ -48,6 +48,7 @@
 #define VTYSH_MAX_ALIAS_DEF_LEN   30
 #define VTYSH_MAX_ALIAS_DEF_LEN_WITH_ARGS   40
 #define VTYSH_MAX_ALIAS_LIST_LEN 400
+#define VTYSH_MAX_PATHNAME_LEN   256
 
 struct vtysh_alias_data {
    char alias_def_str[VTYSH_MAX_ALIAS_DEF_LEN];
diff --git a/vtysh/vtysh_main.c b/vtysh/vtysh_main.c
index 6942b44..001d090 100644
--- a/vtysh/vtysh_main.c
+++ b/vtysh/vtysh_main.c
@@ -298,6 +298,7 @@ main (int argc, char **argv, char **env)
   char *temp_db = NULL;
   pthread_t vtysh_ovsdb_if_thread;
   struct passwd *pw = NULL;
+  char path[VTYSH_MAX_PATHNAME_LEN+1];
 
   /* set CONSOLE as OFF and SYSLOG as DBG for ops-cli VLOG moduler list.*/
   vlog_set_verbosity("CONSOLE:OFF");
@@ -402,7 +403,10 @@ main (int argc, char **argv, char **env)
   vty->type = VTY_SHELL;
   vty->node = VIEW_NODE;
   cmd_init(0);
-  plugins_cli_init(FEATURES_CLI_PATH);
+
+  snprintf(path, sizeof(path), "%s%s", (getenv("OPENSWITCH_INSTALL_PATH") ? getenv("OPENSWITCH_INSTALL_PATH") : ""),
+           FEATURES_CLI_PATH);
+  plugins_cli_init(path);
   ret = pthread_create(&vtysh_ovsdb_if_thread,
                        (pthread_attr_t *)NULL,
                        vtysh_ovsdb_main_thread,
diff --git a/vtysh/vtysh_ovsdb_if.c b/vtysh/vtysh_ovsdb_if.c
index c2da705..888b844 100644
--- a/vtysh/vtysh_ovsdb_if.c
+++ b/vtysh/vtysh_ovsdb_if.c
@@ -228,6 +228,7 @@ ospf_ovsdb_init()
     /* System table */
     ovsdb_idl_add_table(idl, &ovsrec_table_system);
     ovsdb_idl_add_column(idl, &ovsrec_system_col_router_id);
+    ovsdb_idl_add_column(idl, &ovsrec_system_col_cur_cfg);
 
 
     /* OSPF Router */
