diff --git a/src/ops_subagent.c b/src/ops_subagent.c
index 519be70..badd433 100644
--- a/src/ops_subagent.c
+++ b/src/ops_subagent.c
@@ -161,7 +161,10 @@ main (int argc, char **argv) {
   /*  plugins_init is called after OVSDB socket creation
    *  before monitor thread creation.
    */
-  plugins_snmp_init(FEATURE_SNMP_PATH);
+  char plugins_path[256];
+  snprintf(plugins_path, sizeof(plugins_path), "%s" FEATURE_SNMP_PATH,
+           getenv("OPENSWITCH_INSTALL_PATH") ? getenv("OPENSWITCH_INSTALL_PATH") : "");
+  plugins_snmp_init(plugins_path);
 
   /* create a thread to poll OVSDB */
   ret = pthread_create( &snmp_ovsdb_if_thread,
