diff --git a/src/cli/supportability_vty.c b/src/cli/supportability_vty.c
index 9877461..c92261d 100644
--- a/src/cli/supportability_vty.c
+++ b/src/cli/supportability_vty.c
@@ -56,7 +56,9 @@ install_show_vlog()
     char str[MAX_FILENAME_SIZE] = {0,};
     char *ptr = NULL, *cmd = NULL;
     int install = 0;
-    fd = opendir(PID_DIRECTORY);
+    snprintf(str, sizeof(str), "%s" PID_DIRECTORY,
+             getenv("OPENSWITCH_DATA_PATH") ? getenv("OPENSWITCH_DATA_PATH") : "");
+    fd = opendir(str);
     if(fd == NULL) {
         VLOG_ERR("Directory open failure");
         return 1;
@@ -220,7 +222,10 @@ install_show_tech()
         free(help);
         return 1;
     }
-    char* filename="/etc/openswitch/supportability/ops_showtech.yaml";
+    char filename[MAX_FILENAME_SIZE];
+    snprintf(filename, sizeof(filename),
+             "%s/etc/openswitch/supportability/ops_showtech.yaml",
+             (getenv("OPENSWITCH_INSTALL_PATH") ? getenv("OPENSWITCH_INSTALL_PATH") : ""));
     FILE* fh = fopen(filename, "r");
     if (fh == NULL) {
         VLOG_ERR("Failed to open file");
@@ -303,7 +308,10 @@ install_show_evnts()
         free(cmd);
         return 1;
     }
-    FILE* fh = fopen(EVENTS_YAML_FILE, "r");
+    char filename[MAX_FILENAME_SIZE];
+    snprintf(filename, sizeof(filename), "%s" EVENTS_YAML_FILE,
+             (getenv("OPENSWITCH_INSTALL_PATH") ? getenv("OPENSWITCH_INSTALL_PATH") : ""));
+    FILE* fh = fopen(filename, "r");
     if (fh == NULL) {
         VLOG_ERR("Failed to open file");
         free(cmd);
diff --git a/src/eventlog/eventlog.c b/src/eventlog/eventlog.c
index 038208e..a28cfab 100644
--- a/src/eventlog/eventlog.c
+++ b/src/eventlog/eventlog.c
@@ -39,7 +39,7 @@ VLOG_DEFINE_THIS_MODULE(eventlog);
 static event *ev_table = NULL;
 static char *category_table[MAX_CATEGORIES_PER_DAEMON];
 static int category_index = 0;
-
+static char *event_yaml_path = NULL;
 
 /* Function        : strcmp_with_nullcheck
 * Responsibility  : Ensure arguments are not null before calling strcmp
@@ -169,7 +169,17 @@ parse_yaml_for_category(char *category)
     if(category == NULL) {
         return -1;
     }
-    fh = fopen(EVENT_YAML_FILE, "r");
+    if(event_yaml_path == NULL) {
+        char path[512] = {0,};
+        char *envv = (getenv("OPENSWITCH_INSTALL_PATH") ? getenv("OPENSWITCH_INSTALL_PATH") : "");
+        strcpy(path, envv);
+        if ((strlen(path) + strlen(EVENT_YAML_FILE) + 1) > sizeof(path))
+            return -1;
+        strcat(path, EVENT_YAML_FILE);
+        if ((event_yaml_path = strdup(path)) == NULL)
+            return -1;
+    }
+    fh = fopen(event_yaml_path, "r");
     if(fh == NULL) {
         VLOG_ERR("YAML file open failed");
         return -1;
diff --git a/src/featuremapping/feature_mapping.c b/src/featuremapping/feature_mapping.c
index 1a52aae..c339e79 100644
--- a/src/featuremapping/feature_mapping.c
+++ b/src/featuremapping/feature_mapping.c
@@ -29,6 +29,8 @@
 #include <stdlib.h>
 #include "openvswitch/vlog.h"
 
+#define MAX_FILENAME_SIZE 256
+
 VLOG_DEFINE_THIS_MODULE(feature_mapping);
 
 static int
@@ -197,6 +199,7 @@ parse_feature_mapping_yaml(void)
                       break; \
                   }
 
+    char filename[MAX_FILENAME_SIZE];
     FILE *fh=NULL;
     yaml_parser_t parser;
     yaml_event_t  event;
@@ -211,9 +214,11 @@ parse_feature_mapping_yaml(void)
         VLOG_ERR("Failed to initialize parser!");
         return 1;
     }
-    fh = fopen(FEATURE_MAPPING_CONF, "r");
+    snprintf(filename, sizeof(filename), "%s" FEATURE_MAPPING_CONF,
+             (getenv("OPENSWITCH_INSTALL_PATH") ? getenv("OPENSWITCH_INSTALL_PATH") : ""));
+    fh = fopen(filename, "r");
     if(fh == NULL) {
-        VLOG_ERR("Failed to open file :%s",FEATURE_MAPPING_CONF);
+        VLOG_ERR("Failed to open file :%s",filename);
         yaml_parser_delete(&parser);
         return 1;
     }
