diff --git a/cfgdb.py b/cfgdb.py
index 07b2c10..ff406fd 100644
--- a/cfgdb.py
+++ b/cfgdb.py
@@ -30,7 +30,7 @@ idl = None
 def_db = 'unix:/var/run/openvswitch/db.sock'
 
 # OPS_TODO: Need to pull this from the build env
-cfgdb_schema = '/usr/share/openvswitch/configdb.ovsschema'
+cfgdb_schema = "%s/configdb.ovsschema" % ovs.dirs.PKGDATADIR
 
 #Configdb tabe names
 CONFIG_TABLE = "config"
diff --git a/ops_cfgd.py b/ops_cfgd.py
index 64b0a79..36b9174 100644
--- a/ops_cfgd.py
+++ b/ops_cfgd.py
@@ -44,8 +44,8 @@ def_db = 'unix:/var/run/openvswitch/db.sock'
 # Configuration file definitions
 saved_config = None
 # OPS_TODO: Need to pull these three from the build env
-cfgdb_schema = '/usr/share/openvswitch/configdb.ovsschema'
-ovs_schema = '/usr/share/openvswitch/vswitch.ovsschema'
+cfgdb_schema = "%s/configdb.ovsschema" % ovs.dirs.PKGDATADIR
+ovs_schema = "%s/vswitch.ovsschema" % ovs.dirs.PKGDATADIR
 type_startup_config = "startup"
 #3 sec max retry
 max_miliseconds_to_wait_for_config_data = 30
