diff --git a/cfgdb.py b/cfgdb.py
index 07b2c10..fc54119 100644
--- a/cfgdb.py
+++ b/cfgdb.py
@@ -27,10 +27,10 @@ vlog = ovs.vlog.Vlog("cfgdb")
 # ovs definitions
 idl = None
 # OPS_TODO: Need to pull this from the build env
-def_db = 'unix:/var/run/openvswitch/db.sock'
+def_db = "unix:%s/db.sock" % ovs.dirs.RUNDIR
 
 # OPS_TODO: Need to pull this from the build env
-cfgdb_schema = '/usr/share/openvswitch/configdb.ovsschema'
+cfgdb_schema = "%s/configdb.ovsschema" % ovs.dirs.PKGDATADIR
 
 #Configdb tabe names
 CONFIG_TABLE = "config"
diff --git a/cfgdbutil.py b/cfgdbutil.py
index 4d5a2c2..ea93b30 100644
--- a/cfgdbutil.py
+++ b/cfgdbutil.py
@@ -36,7 +36,7 @@ import cfgdb
 type_startup_config = "startup"
 
 vlog = ovs.vlog.Vlog("cfgmgmt")
-TEMPORARY_DB_SHOW_STARTUP = "unix:/var/run/openvswitch/temp_startup.sock"
+TEMPORARY_DB_SHOW_STARTUP = "unix:%s/temp_startup.sock" % ovs.dirs.RUNDIR
 
 
 def show_config(args):
diff --git a/ops_cfgd.py b/ops_cfgd.py
index 64b0a79..919fc4a 100644
--- a/ops_cfgd.py
+++ b/ops_cfgd.py
@@ -39,13 +39,13 @@ from opslib import restparser
 # ovs definitions
 idl = None
 # OPS_TODO: Need to pull this from the build env
-def_db = 'unix:/var/run/openvswitch/db.sock'
+def_db = "unix:%s/db.sock" % ovs.dirs.RUNDIR
 
 # Configuration file definitions
 saved_config = None
 # OPS_TODO: Need to pull these three from the build env
-cfgdb_schema = '/usr/share/openvswitch/configdb.ovsschema'
-ovs_schema = '/usr/share/openvswitch/vswitch.ovsschema'
+cfgdb_schema = "%s/configdb.ovsschema" % ovs.dirs.PKGDATADIR
+ovs_schema = "%s/vswitch.ovsschema" % ovs.dirs.PKGDATADIR
 type_startup_config = "startup"
 #3 sec max retry
 max_miliseconds_to_wait_for_config_data = 30
