diff --git a/opsrest/settings.py b/opsrest/settings.py
index 9687426..ba7a29d 100755
--- a/opsrest/settings.py
+++ b/opsrest/settings.py
@@ -12,8 +12,9 @@
 #  License for the specific language governing permissions and limitations
 #  under the License.
 
-import os.path
+import os
 import custom
+import ovs.dirs
 
 from tornado.options import define
 
@@ -27,12 +28,12 @@ settings['logging'] = 'info'
 settings["static_path"] = os.path.join(os.path.dirname(__file__), "static")
 settings["template_path"] = os.path.join(os.path.dirname(__file__),
                                          "templates")
-settings['ovs_remote'] = 'unix:/var/run/openvswitch/db.sock'
-settings['ovs_schema'] = '/usr/share/openvswitch/vswitch.ovsschema'
-settings['ext_schema'] = '/usr/share/openvswitch/vswitch.extschema'
+settings['ovs_remote'] = "unix:%s/db.sock" % ovs.dirs.RUNDIR
+settings['ovs_schema'] = "%s/vswitch.ovsschema" % ovs.dirs.PKGDATADIR
+settings['ext_schema'] = "%s/vswitch.extschema" % ovs.dirs.PKGDATADIR
 settings['auth_enabled'] = True
-settings['cfg_db_schema'] = '/usr/share/openvswitch/configdb.ovsschema'
+settings['cfg_db_schema'] = "%s/configdb.ovsschema" % ovs.dirs.PKGDATADIR
 
 settings["account_schema"] = os.path.join(os.path.dirname(custom.__file__),
                                           'schemas/Account.json')
-settings["passwd_srv_yaml"] = '/etc/ops-passwd-srv/ops-passwd-srv.yaml'
+settings["passwd_srv_yaml"] = "%s/etc/ops-passwd-srv/ops-passwd-srv.yaml" % os.environ.get("OPENSWITCH_INSTALL_PATH", "")
diff --git a/opsrest/urls.py b/opsrest/urls.py
index 5cbab5f..83c23db 100755
--- a/opsrest/urls.py
+++ b/opsrest/urls.py
@@ -21,6 +21,7 @@ from opsrest.handlers.websocket.notifications import WSNotificationsHandler
 from custom.logcontroller import LogController
 from custom.accountcontroller import AccountController
 from custom.configcontroller import ConfigController
+import os
 
 REGEX_RESOURCE_ID = '?(?P<resource_id>[A-Za-z0-9-_]+[$]?)?/?'
 
@@ -43,6 +44,6 @@ custom_url_patterns =\
 
 static_url_patterns =\
     [(r"/api/(.*)", StaticContentHandler,
-     {"path": "/srv/www/api", "default_filename": "index.html"}),
+     {"path": "%s/srv/www/api" % os.environ.get("OPENSWITCH_INSTALL_PATH", ""), "default_filename": "index.html"}),
      (r"/(.*)", StaticContentHandler,
-     {"path": "/srv/www/static", "default_filename": "index.html"})]
+      {"path": "%s/srv/www/static" % os.environ.get("OPENSWITCH_INSTALL_PATH", ""), "default_filename": "index.html"})]
diff --git a/restd.py b/restd.py
index 961a79d..28e4522 100644
--- a/restd.py
+++ b/restd.py
@@ -31,14 +31,16 @@ from tornado.log import enable_pretty_logging
 options.logging = settings['logging']
 enable_pretty_logging()
 
-SSL_PRIV_DIR = "/etc/ssl/private"
-SSL_PRIV_KEY_FILE = "/etc/ssl/private/server-private.key"
-SSL_CRT_FILE = "/etc/ssl/certs/server.crt"
-
+SSL_PRIV_DIR = "%s/etc/ssl/private" % os.environ.get("OPENSWITCH_DATA_PATH", "")
+SSL_PRIV_KEY_FILE = "%s/etc/ssl/certs/server-private.key" % os.environ.get("OPENSWITCH_DATA_PATH", "")
+SSL_CRT_DIR = "%s/etc/ssl/certs" % os.environ.get("OPENSWITCH_DATA_PATH", "")
+SSL_CRT_FILE = "%s/etc/ssl/certs/server.crt" % os.environ.get("OPENSWITCH_DATA_PATH", "")
 
 def create_ssl_pki():
     if not os.path.exists(SSL_PRIV_DIR):
-        os.mkdir(SSL_PRIV_DIR, 0700)
+        os.makedirs(SSL_PRIV_DIR, 0700)
+    if not os.path.exists(SSL_CRT_DIR):
+        os.makedirs(SSL_CRT_DIR, 0700)
 
     if os.path.isfile(SSL_CRT_FILE) and os.path.isfile(SSL_PRIV_KEY_FILE):
         # Create these files only once on system bootup.
